here-naksha-app-service/src/test/java/com/here/naksha/app/service/ReadFeaturesByRadiusPostTest.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.app.service;
+
+import com.here.naksha.app.common.ApiTest;
+import com.here.naksha.app.common.NakshaTestWebClient;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+import static com.here.naksha.app.common.CommonApiTestSetup.setupSpaceAndRelatedResources;
+import static com.here.naksha.app.common.TestUtil.loadFileOrFail;
+import static com.here.naksha.app.common.assertions.ResponseAssertions.assertThat;
+
+class ReadFeaturesByRadiusPostTest extends ApiTest {
+
+  private static final NakshaTestWebClient nakshaClient = new NakshaTestWebClient();
+
+  private static final String SPACE_ID = "read_features_by_radius_post_test_space";
+
+  /*
+  For this test suite, we upfront create various Features using different combination of Geometry, Tags and properties.
+  And then in subsequent tests, we validate the REST API behaviour by providing different filter conditions
+
+  Main Space:
+    - feature 1 - Point1, Tag-1, Tag-Ref, Property-1
+    - feature 2 - Point2-within-5m-of-Point1, Tag-2, Tag-Ref, Property-2
+    - feature 3 - Point2-within-5m-of-Point1, Tag-3, Tag-Ref, Property-1, Property-2
+    - feature 4 - Point3-outside-5m-of-Point1, all above tags, all above properties
+
+  Test Cases:
+    TC  1 - Point1, radius=0 (should return feature 1 only)
+    TC  2 - Point1, radius=5m (should return features 1,2,3)
+    TC  3 - Point1, radius=5m, Prop-2 (should return features 2,3)
+    TC  4 - Point1, radius=5m, Tag-1 (should return feature 1 only)
+    TC  5 - Point1, radius=5m, Prop-1, Tag-3 (should return feature 3 only)
+    TC  6 - Point1, radius=5m, Limit-2 (should return features 1,2)
```

Comment:
I remember Result may not guarantee any ordering of its content right? Can we make sure here that the first 2 features returned is always 1 and 2, and not 3? This is also the same with the other GetByRadius API previously

Reply:
Ok, while limit has been in place for previous APIs as well and unit tests have always worked with correct order against a newly created schema and tables as part of test setup, I agree that it is a fair point that it may not be guaranteed to continue working like that in future (eventhough chances are less).
I have fixed both the tests, GetByRadius and GetByRadiusPost